packages/ytube/src/lib/youtubeClient.ts

```diff
@@ -1,202 +1,293 @@
-        R.otherwise((err) => Result.Error<User, DomainError>(new YoutubeAuthorizationFailed(err.message)))
-      )
-      return createUserFlow(code);
-  }
-
-  getChannels = async (user: User) => {
-    const result = await R.pipe(
-      (u: User) => this.getYoutube(u.accessToken, u.refreshToken),
-      yt => Result.tryAsync(() => yt.channels.list({
-        part: ['snippet', 'contentDetails', 'statistics'],
-        mine: true,
-      }), new DomainError('Failed to retrieve channels list for')),
-      R.andThen(channelsResult => channelsResult.map(c => this.mapChannels(user, c.data.items ?? []))),
-    )(user)
-    return result;
-  }
-  getVideos = (channel: Channel, top: number) => {
-    return R.pipe(
-      () => this.getYoutube(channel.userAccessToken, channel.userRefreshToken),
-      yt => Result.tryAsync(() => this.iterateVideos(yt, channel, top), new DomainError(`Failed to fetch videos for channel ${channel.title}`)),
-    )()
-  }
-  getAllVideos = (channel: Channel, max = 500) => {
-    return R.pipe(
-      () => this.getYoutube(channel.userAccessToken, channel.userRefreshToken),
-      yt => Result.tryAsync(() => this.iterateVideos(yt, channel, max), new DomainError(`Failed to fetch videos for channel ${channel.title}`))
-    )()
-  }
-  downloadVideo = (videoUrl: string): Readable => {
-    return ytdl(videoUrl);
-  }
-  private async iterateVideos(youtube: youtube_v3.Youtube, channel: Channel, max: number){
-    let videos: Video[] = [];
-    let continuation: string;
+    const tokenResponse = await this.getAccessToken(code)
+    const tokenInfo = await this.getAuth().getTokenInfo(tokenResponse.tokens.access_token)
+
+    const user = new User(
+      tokenInfo.sub,
+      tokenInfo.email,
+      tokenResponse.tokens.access_token,
+      tokenResponse.tokens.refresh_token
+    )
+    return user
+  }
+
+  async getChannels(user: User) {
+    const yt = this.getYoutube(user.accessToken, user.refreshToken)
+
+    const channelResponse = await yt.channels.list({
+      part: ['snippet', 'contentDetails', 'statistics'],
+      mine: true,
+    })
+    const channels = this.mapChannels(user, channelResponse.data.items ?? [])
+
+    return channels
+  }
+
+  async verifyChannel(channel: Channel): Promise<Channel> {
+    const errors: ChannelVerificationFailed[] = []
+    if (channel.statistics.subscriberCount < MINIMUM_SUBSCRIBERS_COUNT) {
+      errors.push({
+        errorCode: ExitCodes.CHANNEL_CRITERIA_UNMET_SUBSCRIBERS,
+        message:
+          `Channel ${channel.id} with ${channel.statistics.subscriberCount} subscribers does not ` +
+          `meet Youtube Partner Program requirement of ${MINIMUM_SUBSCRIBERS_COUNT} subscribers`,
+        result: channel.statistics.subscriberCount,
+        expected: MINIMUM_SUBSCRIBERS_COUNT,
+      })
+    }
+
+    // at least MINiMUM_VIDEO_COUNT videos should be one month old
+    const oneMonthsAgo = new Date()
+    oneMonthsAgo.setMonth(oneMonthsAgo.getMonth() - 3)
```

Comment:
This talks about one month old but you actually subtract 3 months. Maybe we store both this (MINIMUM_VIDEO_AGE_MONTHS=1) and MINIMUM_CHANNEL_AGE_MONTHS=3 below as const variables up top for better visibility?